overlord/snapstate/backend/link.go

```diff
@@ -411,21 +409,17 @@ func (b Backend) UnlinkComponent(cpi snap.ContainerPlaceInfo, snapRev snap.Revis
 	return nil
 }
 
-func (b Backend) KillSnapApps(snapName string, reason snap.AppKillReason, meter progress.Meter, tm timings.Measurer) error {
-	cli := client.New()
-
-	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(timeout.DefaultTimeout))
-	defer cancel()
+func (b Backend) KillSnapApps(snapName string, reason snap.AppKillReason, tm timings.Measurer) error {
+	if reason != snap.KillReasonOther {
+		logger.Debugf("KillSnapApps called for %q, reason: %v", snapName, reason)
+	} else {
+		logger.Debugf("KillSnapApps called for %q", snapName)
+	}
 
 	var err error
-	var failures []client.AppFailure
 	timings.Run(tm, "kill-snap-apps", fmt.Sprintf("kill running apps for snap %s", snapName), func(timings.Measurer) {
-		failures, err = cli.AppsKill(ctx, []string{snapName}, syscall.SIGKILL, reason)
+		err = cgroupKillSnapProcesses(context.Background(), snapName)
```

Comment:
I wonder if we should use a `context.TODO` and perhaps bubble it up somehow. It's also interesting, from task engine perspecitve, that tasks might want to allow to be cancelled this way.

Reply:
Totally, Added a TODO